armi/nuclearDataIO/rzflux.py

```diff
@@ -0,0 +1,186 @@
+    "TNBL",
+    "TNBAL",
+    "TNCRA",
+    "X1",
+    "X2",
+    "X3",
+    "NBLOK",
+    "ITPS",
+    "NZONE",
+    "NGROUP",
+    "NCY",
+)
+
+IMPLICIT_INT = "IJKLMN"
+
+
+class RzfluxData:
+    """
+    Data representation that can be read from or written to a RZFLUX file.
+
+    Notes
+    -----
+    Analogous to a IsotxsLibrary for ISTOXS files.
+    """
+
+    def __init__(self):
+        # Need Metadata subclass for default keys
+        self.metadata = nuclearFileMetadata._Metadata()
+
+        # 2D data
+        self.groupFluxes = None
+
+
+class RzfluxStream(cccc.Stream):
+    """
+    Stream for reading to/writing from with RZFLUX data.
+
+    Parameters
+    ----------
+    flux : RzfluxData
+        Data structure
+    fileName: str
+        path to RZFLUX file
+    fileMode: str
+        string indicating if ``fileName`` is being read or written, and
+        in ascii or binary format
+
+    """
+
+    def __init__(self, flux: RzfluxData, fileName: str, fileMode: str):
+        cccc.Stream.__init__(self, fileName, fileMode)
+        self.fc = {}  # file control info (sort of global for this library)
+        self._flux = flux
+        self._metadata = self._getFileMetadata()
+
+    def _getFileMetadata(self):
+        return self._flux.metadata
+
+    @classmethod
+    def _read(cls, fileName: str, fileMode: str) -> RzfluxData:
+        flux = RzfluxData()
+        return cls._readWrite(
+            flux,
+            fileName,
+            fileMode,
+        )
+
+    # pylint: disable=arguments-differ
+    @classmethod
+    def _write(cls, flux: RzfluxData, fileName: str, fileMode: str):
+        return cls._readWrite(flux, fileName, fileMode)
+
+    @classmethod
+    def _readWrite(cls, flux: RzfluxData, fileName: str, fileMode: str) -> RzfluxData:
+        with RzfluxStream(flux, fileName, fileMode) as rw:
+            rw.readWrite()
+        return flux
+
+    def readWrite(self):
+        """
+        Step through the structure of the file and read/write it.
+        """
+        self._rwFileID()
+        self._rw1DRecord()
+        self._rw2DRecord()
+
+    def _rwFileID(self):
+        """
+        Read/write file id record.
+
+        Notes
+        -----
+        The username, version, etc are embedded in this string but it's
+        usually blank. The number 28 was actually obtained from
+        a hex editor and may be code specific.
+        """
+        with self.createRecord() as record:
+            self._metadata["label"] = record.rwString(self._metadata["label"], 28)
+
+    def _rw1DRecord(self):
+        """
+        Read/write File specifications on 1D record.
+
+        This record contains 27 integers.
+        """
+        with self.createRecord() as record:
+            for key in FILE_SPEC_1D_KEYS:
+                # ready for some implicit madness from the FORTRAN 77 days?
+                if key[0] in IMPLICIT_INT:
+                    self._metadata[key] = record.rwInt(self._metadata[key])
```

Comment:
anything that starts with I, J, K, N, M, etc. is an integer so it's not just ITPS. Even EFFK has to be spelled that way to be a float because keff is implicitly an int!

This is what happens when you *don't* use `implicit none` 😱